### PR TITLE
reopen How-to run qsvt on hardware

### PR DESCRIPTION
**Pull request by KetpuntoG**
**Title:** how to run QSVT on Hardware

**Summary:** Performing polynomial transformations on matrices or operators is a task of great interest, which we have previously addressed in our introductory demos on QSVT and its practical applications. In this how-to guide, we will show how we can implement the QSVT subroutine in a hardware-compatible way, taking your applications to the next level.

----
If you are writing a demonstration, please answer these questions to facilitate the marketing process.

* GOALS — Why are we working on this now?

We are making a Push to QSVT as it currently gives many problems and we want to make clear how to use it correctly.

* AUDIENCE — Who is this for?

People who know the algorithm but want to see how to implement it in hardware.

* KEYWORDS — What words should be included in the marketing post?

QSVT, hardware, how-to

* Which of the following types of documentation is most similar to your file? 

    
- [ ] Tutorial
- [ ] Demo
- [ x ] How-to

### CODE REVIEW

File: demonstrations/tutorial_qsvt_hardware.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+#
+# .. [#unification]
+#
+#     John M. Martyn, Zane M. Rossi, Andrew K. Tan, Isaac L. Chuang.
+#     "A Grand Unification of Quantum Algorithms".
+#     `arXiv preprint arXiv:2105.02859 <https://arxiv.org/abs/2105.02859>`__.
+#
+# About the author
+# ----------------
+#
```
Comment (maliasadi) [inline demonstrations/tutorial_qsvt_hardware.py:161]: To fix the rendering issue:
```suggestion
#

```

**Review by maliasadi — APPROVED** [review]
:partying_face:

**Comment by ikurecic** [comment]
[sc-67593]

File: demonstrations/tutorial_qsvt_hardware.metadata.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        {
+            "id": "guillermo_alonso"
+        }
+    ],
+    "dateOfPublication": "2023-05-23T00:00:00+00:00",
+    "dateOfLastModification": "2024-03-04T00:00:00+00:00",
+    "categories": [
+        "Algorithms",
+        "Quantum Computing"
+    ],
```
Comment (ikurecic) [inline demonstrations/tutorial_qsvt_hardware.metadata.json:13]: Should this also have the 'How-to' category?

File: demonstrations/tutorial_qsvt_hardware.metadata.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{
+    "title": "How to implement QSVT on hardware",
+    "authors": [
+        {
+            "id": "guillermo_alonso"
```
Comment (ikurecic) [inline demonstrations/tutorial_qsvt_hardware.metadata.json:5]:
```suggestion
            "username": "KetPuntoG"
```

File: demonstrations/tutorial_qsvt_hardware.metadata.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        }
+    ],
+    "dateOfPublication": "2023-05-23T00:00:00+00:00",
+    "dateOfLastModification": "2024-03-04T00:00:00+00:00",
```
Comment (ikurecic) [inline demonstrations/tutorial_qsvt_hardware.metadata.json:9]: tk

File: demonstrations/tutorial_qsvt_hardware.metadata.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    "previewImages": [
+        {
+            "type": "thumbnail",
+            "uri": "/_static/demonstration_assets/intro_qsvt/thumbnail_intro_qsvt.png"
+        },
+        {
+            "type": "large_thumbnail",
+            "uri": "/_static/demo_thumbnails/large_demo_thumbnails/thumbnail_large_intro_qsvt.png"
+        }
+    ],
```
Comment (ikurecic) [inline demonstrations/tutorial_qsvt_hardware.metadata.json:24]: tk.

File: demonstrations/tutorial_qsvt_hardware.py [hunk 6]
```diff
@@ -0,0 +1,161 @@
+r"""How to implement QSVT on hardware
+=======================================
+
+The :doc:`Quantum Singular Value Transform (QSVT) </demos/tutorial_intro_qsvt>`
```
Comment (ikurecic) [inline demonstrations/tutorial_qsvt_hardware.py:4]:
```suggestion
The :doc:`quantum singular value transform (QSVT) </demos/tutorial_intro_qsvt>`
```

File: demonstrations/tutorial_qsvt_hardware.py [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+is a quantum algorithm that allows us to perform polynomial
+transformations on matrices or operators, and is rapidly becoming
```
Comment (ikurecic) [inline demonstrations/tutorial_qsvt_hardware.py:6]:
```suggestion
transformations on matrices or operators, and it is rapidly becoming
```

File: demonstrations/tutorial_qsvt_hardware.py [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+to the next level.
+
+Angles calculation
+------------------
+
+Our goal is to apply a polynomial transformation to a given Hamiltonian, i.e., :math:`p(\mathcal{H})`. To achieve this, we must consider the two
+fundamental components of the QSVT algorithm:
+
+- **Projection angles**: A list of angles that will determine the coefficients of the polynomial to be applied.
+- **Block Encoding**: The strategy used to encode the Hamiltonian. We will use the :doc:`Linear Combinations of Unitaries <demos/tutorial_lcu_blockencoding>` approach via the PennyLane :class:`~.qml.PrepSelPrep` operation.
```
Comment (ikurecic) [inline demonstrations/tutorial_qsvt_hardware.py:21]:
```suggestion
- **Block encoding**: The strategy used to encode the Hamiltonian. We will use the :doc:`linear combinations of unitaries <demos/tutorial_lcu_blockencoding>` approach via the PennyLane :class:`~.qml.PrepSelPrep` operation.
```

File: demonstrations/tutorial_qsvt_hardware.py [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+transformations on matrices or operators, and is rapidly becoming
+a go-to algorithm for :doc:`quantum application research </demos/tutorial_apply_qsvt>`
+in the `ISQ era <https://pennylane.ai/blog/2023/06/from-nisq-to-isq/>`__.
+
+In this how-to guide, we will show how we can implement the QSVT
+subroutine in a hardware-compatible way, taking your application research
+to the next level.
+
+Angles calculation
+------------------
```
Comment (ikurecic) [inline demonstrations/tutorial_qsvt_hardware.py:15]:
```suggestion
Calculating angles
------------------
```
It simplifies it a bit.

File: demonstrations/tutorial_qsvt_hardware.py [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+
+Our goal is to apply a polynomial transformation to a given Hamiltonian, i.e., :math:`p(\mathcal{H})`. To achieve this, we must consider the two
+fundamental components of the QSVT algorithm:
+
+- **Projection angles**: A list of angles that will determine the coefficients of the polynomial to be applied.
+- **Block Encoding**: The strategy used to encode the Hamiltonian. We will use the :doc:`Linear Combinations of Unitaries <demos/tutorial_lcu_blockencoding>` approach via the PennyLane :class:`~.qml.PrepSelPrep` operation.
+
+Calculating angles is not a trivial task but there are tools such as `pyqsp <https://github.com/ichuang/pyqsp/tree/master/pyqsp>`_ that do the job for us.
```
Comment (ikurecic) [inline demonstrations/tutorial_qsvt_hardware.py:23]:
```suggestion
Calculating angles is not a trivial task, but there are tools such as `pyqsp <https://github.com/ichuang/pyqsp/tree/master/pyqsp>`_ that do the job for us.
```

File: demonstrations/tutorial_qsvt_hardware.py [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+    return angles + update_vals
+
+
+angles = convert_angles(ang_seq)
+print(angles)
+
+######################################################################
+# Using these angles, we can now start working with the template.
+#
+# QSVT on Hardware
```
Comment (ikurecic) [inline demonstrations/tutorial_qsvt_hardware.py:76]:
```suggestion
# QSVT on hardware
```

**Review by ikurecic — COMMENTED** [review]
Nice, @KetpuntoG , thanks.
I left a couple of minor comments/suggestions.

File: demonstrations/tutorial_qsvt_hardware.metadata.json (hunk 2 above)
Comment (KetpuntoG) [reply demonstrations/tutorial_qsvt_hardware.metadata.json:13]: Sure 🙌

**Review by ikurecic — APPROVED** [review]

**Pull request merged**